frame/statement/src/lib.rs

```diff
@@ -0,0 +1,214 @@
+					Some(e) => {
+						statement.strip_proof();
+						if let Ok(Event::NewStatement { account: a, statement: s }) = e.try_into() {
+							if a != account || s != statement {
+								log::debug!(target: LOG_TARGET, "Event data mismatch");
+								return Err(InvalidStatement::BadProof)
+							}
+						} else {
+							log::debug!(target: LOG_TARGET, "Event type mismatch");
+							return Err(InvalidStatement::BadProof)
+						}
+					},
+					_ => {
+						log::debug!(target: LOG_TARGET, "Bad event index");
+						return Err(InvalidStatement::BadProof)
+					},
+				}
+				account
+			},
+			_ => match statement.verify_signature() {
+				SignatureVerificationResult::Valid(account) => account.into(),
+				SignatureVerificationResult::Invalid => {
+					log::debug!(target: LOG_TARGET, "Bad statement signature.");
+					return Err(InvalidStatement::BadProof)
+				},
+				SignatureVerificationResult::NoSignature => {
+					log::debug!(target: LOG_TARGET, "Missing statement signature.");
+					return Err(InvalidStatement::NoProof)
+				},
+			},
+		};
+		let statement_cost = T::StatementCost::get();
+		let byte_cost = T::ByteCost::get();
+		let priority_cost = statement_cost;
+		let balance = <T::Currency as Inspect<AccountIdOf<T>>>::balance(&account);
+		let global_priority =
+			balance.checked_div(&priority_cost).unwrap_or_default().saturated_into();
+		let max_count = balance
+			.checked_div(&statement_cost)
+			.unwrap_or_default()
+			.saturated_into::<u32>()
+			.clamp(MIN_ALLOWED_STATEMENTS, MAX_ALLOWED_STATEMENTS);
+		let max_size = balance
+			.checked_div(&byte_cost)
+			.unwrap_or_default()
+			.saturated_into::<u32>()
+			.clamp(MIN_ALLOWED_BYTES, MAX_ALLOWED_BYTES);
+
+		Ok(ValidStatement { global_priority, max_count, max_size })
+	}
+
+	/// Submit a statement event. The statement will be picked up by the offchain worker and
+	/// broadcast to the network.
+	pub fn submit_statement(account: T::AccountId, statement: Statement) {
+		Self::deposit_event(Event::NewStatement { account, statement });
+	}
+
+	fn collect_statements() {
+		// Find `NewStatement` events and submit them to the store
+		for (index, event) in frame_system::Pallet::<T>::read_events_no_consensus().enumerate() {
+			if let Ok(Event::<T>::NewStatement { account, mut statement }) = event.event.try_into()
+			{
+				if statement.proof().is_none() {
+					let proof = Proof::OnChain {
+						who: account.into(),
+						block_hash: frame_system::Pallet::<T>::parent_hash().into(),
+						event_index: index as u64,
+					};
+					statement.set_proof(proof);
+				}
+				sp_statement_store::runtime_api::io::submit_statement(statement);
+			}
+		}
+	}
+
+	fn dispatch_statements() {
+		let all_statements = sp_statement_store::runtime_api::io::dump();
+		for (hash, _statement) in all_statements {
+			// TODO: Custom statement handling
+			log::trace!(target: LOG_TARGET, "Handling statement #{:?}", hash);
+		}
```

Comment:
What is the idea with this?

Reply:
The OCW should be exteneded later to do something useful with the statements that are in the store.

Reply:
I'll remove it for now. It will prbably be in a custom pallet anyway.